### PR TITLE
Fix paged block reader transfer offset

### PR DESCRIPTION
**Pull request by dbw9580**
Fix incorrect initial offset in paged block reader.

### CODE REVIEW

File: core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java [hunk 1]
```diff
@@ -61,19 +61,25 @@ public class PagedBlockReader extends BlockReader {
...
+   * @param offset initial offset within the block to begin the read from
    * @param ufsBlockOptions options to open a ufs block
    */
   public PagedBlockReader(CacheManager cacheManager, UfsManager ufsManager,
                           UfsInputStreamCache ufsInStreamCache,
                           AlluxioConfiguration conf,
-                          long blockId, Protocol.OpenUfsBlockOptions ufsBlockOptions) {
-
+                          long blockId, long offset, Protocol.OpenUfsBlockOptions ufsBlockOptions) {
+    if (offset < 0 || offset > ufsBlockOptions.getBlockSize()) {
```
Comment (beinan) [inline core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java:71]: How about using IllegalArgumentException?  I think IndexOutOfBoundsException is designed to array or string.

I might prefer to use Preconditions.checkArgument in this case

**Review by beinan — COMMENTED** [review]
looks good to me except a minor issue

File: core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java (hunk 1 above)
Comment (jja725) [reply core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java:71]: how about OutOfRangeRuntimeException or InternalRuntimeException? I would personally think IndexOutofBounds is more aligned with the original java API but here if we pass in the wrong parameter then it's an internal code bug so Internal Exception makes sense as well.

Comment (dbw9580) [reply core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java]: Changed to use `Preconditions.checkArgument`.

File: core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java [hunk 2]
```diff
@@ -164,6 +168,11 @@ public ReadableByteChannel getChannel() {
   @Override
   public int transferTo(ByteBuf buf) throws IOException {
     Preconditions.checkState(!mClosed);
+    // todo(bowen): mUfsBlockOptions may not always carry blockSize. get correct size from
+    // block metastore
```
Comment (dbw9580) [inline core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java:172]: @beinan Please do not merge this until #16157 is merged, so I can address this based on changes in that PR.

Comment (beinan) [reply core/server/worker/src/main/java/alluxio/worker/page/PagedBlockReader.java:172]: just merged #16157

**Review by beinan — APPROVED** [review]
lgtm, waiting for the ci

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Pull request merged**